Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystemEditorOperations.cs

```diff
@@ -641,40 +655,42 @@ private void EditorUpdateBuildSettings()
                 return;
             }
 
-            bool changedScenes = false;
-
             if (profile.UseManagerScene)
             {
-                changedScenes |= EditorSceneUtils.AddSceneToBuildSettings(
-                    profile.ManagerScene, 
-                    cachedBuildScenes, 
-                    EditorSceneUtils.BuildIndexTarget.First);
+                if (EditorSceneUtils.AddSceneToBuildSettings(
+                    profile.ManagerScene,
+                    cachedBuildScenes,
+                    EditorSceneUtils.BuildIndexTarget.First))
+                {
+                    cachedBuildScenes = EditorBuildSettings.scenes;
+                }
             }
 
             foreach (SceneInfo contentScene in profile.ContentScenes)
             {
-                changedScenes |= EditorSceneUtils.AddSceneToBuildSettings(
+                if (EditorSceneUtils.AddSceneToBuildSettings(
                     contentScene,
                     cachedBuildScenes, 
-                    EditorSceneUtils.BuildIndexTarget.None);
+                    EditorSceneUtils.BuildIndexTarget.None))
+                {
+                    cachedBuildScenes = EditorBuildSettings.scenes;
```

Comment:
>cachedBuildScenes = EditorBuildSettings.scenes; [](start = 20, length = 47)

The previous thing would set the cached build scenes at the very end (instead of setting it on each iteration) - curious to know about why we went from reconciling/snapping the state of the array at the end, vs incrementally building it (and thus, calling it multiple times as it gets increasingly larger)

Reply:
This was working fine if only one scene was being added at a time, but it would get confused if multiple scenes were being added. Later parts of the loop would be working off old data.